ArjunaJTA/cdi/tests/classes/com/hp/mwtests/ts/jta/cdi/transactional/stereotype/extension/AnnotatedTypeWrapper.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2017, Red Hat Middleware LLC, and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package com.hp.mwtests.ts.jta.cdi.transactional.stereotype.extension;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.enterprise.inject.spi.AnnotatedConstructor;
+import javax.enterprise.inject.spi.AnnotatedField;
+import javax.enterprise.inject.spi.AnnotatedMethod;
+import javax.enterprise.inject.spi.AnnotatedType;
+import javax.enterprise.inject.spi.ProcessAnnotatedType;
+
+public class AnnotatedTypeWrapper<T> implements AnnotatedType<T> {
+
+    private final AnnotatedType<T> wrapped;
+    private final Set<Annotation> annotations;
+
+    public AnnotatedTypeWrapper(AnnotatedType<T> wrapped, Set<Annotation> annotations) {
+        this.wrapped = wrapped;
+        this.annotations = new HashSet<>(annotations);
+    }
+
+    public AnnotatedTypeWrapper(ProcessAnnotatedType<T> processAnnType) {
+        this(processAnnType.getAnnotatedType(), processAnnType.getAnnotatedType().getAnnotations());
+    }
+
+    public void addAnnotation(Annotation annotation) {
+        annotations.add(annotation);
+    }
+
+    public void addAnnotation(Class<? extends Annotation> classAnnotation) {
+        Annotation annotation = new Annotation() {
+            @Override
+            public Class<? extends Annotation> annotationType() {
+                return classAnnotation;
+            }
+        };
+        addAnnotation(annotation);
+    }
+
+    @Override
+    public <A extends Annotation> A getAnnotation(Class<A> annotationType) {
+        return wrapped.getAnnotation(annotationType);
```

Comment:
You should use the `annotations` field here too, not the `wrapped` reference.

Reply:
true, thanks